scripts/remove-idle-users.py

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env python
+from __future__ import print_function
+import argparse
+import sys
+import json
+import urllib
+import requests
+
+def get_room_id(homeserver, alias, token):
+    res = requests.get(homeserver + "/_matrix/client/r0/directory/room/" + urllib.quote(alias) + "?access_token=" + token)
+    res.raise_for_status()
+    return res.json()["room_id"]
+
+def get_last_active_ago(homeserver, user_id, token):
+    res = requests.get(homeserver + "/_matrix/client/r0/presence/" + urllib.quote(user_id) + "/status?access_token=" + token).json()
+    return res.get("last_active_ago", None)
+
+def is_idle(homeserver, user_id, token, activity_threshold_ms):
+    return get_last_active_ago(homeserver, user_id, token) > activity_threshold_ms
+
+def get_idle_users(homeserver, room_id, token, since):
+    res = requests.get(homeserver + "/_matrix/client/r0/rooms/" + urllib.quote(room_id) + "/joined_members?access_token=" + token)
+    user_ids = [user_id for user_id in res.json().get("joined", None)]
+
+    activity_threshold_ms = since * 24 * 60 * 60 * 1000
+
+    return [user_id for user_id in user_ids if is_idle(homeserver, user_id, token, activity_threshold_ms)]
+
+def kick_idlers(homeserver, homeserver_domain, room_id, token, since, user_prefix):
+    reason = "Being idle for >%s days" % since
+
+    user_ids = get_idle_users(homeserver, room_id, token, since)
+    failure_responses = []
+    count = 0
+    print("There are %s idle users in %s" % (len(user_ids), room_id))
+    for user_id in user_ids:
+        # Ignore users that do not start with the user_prefix
+        if not user_id.startswith(user_prefix):
+            continue
+        res = requests.post(
+            homeserver + "/_matrix/client/r0/rooms/" + urllib.quote(room_id) + "/kick?access_token=" + token,
+            data = json.dumps({
+                "reason": reason,
```

Comment:
If only it were that easy. Whilst the [spec](http://matrix.org/docs/spec/client_server/r0.2.0.html#post-matrix-client-r0-rooms-roomid-kick) states that `reason` is a valid key, it looks like [synapse](https://github.com/matrix-org/synapse/blob/master/synapse/rest/client/v1/room.py#L564) doesn't abide by this.

Use [this API](http://matrix.org/docs/spec/client_server/r0.2.0.html#put-matrix-client-r0-rooms-roomid-state-eventtype-statekey) instead. Will file an issue with Synapse peeps.